sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraServiceImplTest.java

```diff
@@ -0,0 +1,92 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.cassandra;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Metadata;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Tests on {@link CassandraServiceImplTest}.
+ */
+public class CassandraServiceImplTest {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CassandraServiceImplTest.class);
+
+  private static final String MURMUR3_PARTITIONER = "org.apache.cassandra.dht.Murmur3Partitioner";
+
+  private Cluster createClusterMock() {
+    Metadata metadata = Mockito.mock(Metadata.class);
+    Mockito.when(metadata.getPartitioner()).thenReturn(MURMUR3_PARTITIONER);
+    Cluster cluster = Mockito.mock(Cluster.class);
+    Mockito.when(cluster.getMetadata()).thenReturn(metadata);
+    return cluster;
+  }
+
+  @Test
+  public void testDistanceWithMurmur3Partitioner() throws Exception {
+    long distance = CassandraServiceImpl.distance(10L, 100L);
+    assertEquals(90, distance);
+  }
+
+  @Test
+  public void testValidPartitioner() throws Exception {
+    assertTrue(CassandraServiceImpl.isMurmur3Partitioner(createClusterMock()));
+  }
+
+  @Test
+  public void testEstimatedSizeBytes() throws Exception {
+    List<CassandraServiceImpl.TokenRange> tokenRanges = new ArrayList<>();
+    tokenRanges.add(new CassandraServiceImpl.TokenRange(0, 10000000, 0, 99999999));
+    tokenRanges.add(new CassandraServiceImpl.TokenRange(1, 10000000, 100000000, 199999999));
+    tokenRanges.add(new CassandraServiceImpl.TokenRange(2, 10000000, 200000000, 299999999));
+    tokenRanges.add(new CassandraServiceImpl.TokenRange(3, 10000000, 300000000, 399999999));
+    tokenRanges.add(new CassandraServiceImpl.TokenRange(4, 10000000, 400000000, 499999999));
+    assertEquals(100000000, CassandraServiceImpl.getEstimatedSizeBytes(tokenRanges));
+  }
+
+  @Test
+  public void testSplitIntoBundles() throws Exception {
+    CassandraServiceImpl service = new CassandraServiceImpl();
+    CassandraIO.Read spec = CassandraIO.read().withKeyspace("beam").withTable("test");
+    List<CassandraIO.CassandraSource> sources = service.split(spec, 1000, 10000);
+    assertEquals(10, sources.size());
+    for (CassandraIO.CassandraSource source : sources) {
```

Comment:
This test will not catch any but the most egregious possible bugs in implementation of `split`, if it verifies only the number of splits. `split` is a correctness-critical method and its result should be verified more thoroughly - please verify that the actual produced queries are correct.